src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -503,6 +513,35 @@ void IslandWindow::SetAlwaysOnTop(const bool alwaysOnTop)
     }
 }
 
+// Method Description:
+// - Sets the taskbar progress indicator
+// Arguments:
+// - state: indicates the progress state
+// - progress: indicates the progress value
+void IslandWindow::SetTaskbarProgress(const size_t state, const size_t progress)
+{
+    if (_taskbarInitialized)
+    {
+        switch (state)
+        {
+        case 0:
+            // removes the taskbar progress indicator
+            _taskbar->SetProgressState(_window.get(), TBPF_NOPROGRESS);
+            break;
+        case 1:
+            // sets the progress value to value given by the 'progress' parameter
+            _taskbar->SetProgressValue(_window.get(), progress, 100);
+            break;
+        case 2:

```

Comment:
When we're implementing this, please make sure to also include the undocumented `3` state for `TBPF_INDETERMINATE` (see https://github.com/microsoft/terminal/issues/3004#issuecomment-716894025)